test/functorch_additional_op_db.py

```diff
@@ -342,3 +341,36 @@ def sample_inputs_binary_cross_entropy(op_info, device, dtype, requires_grad, lo
         supports_out=False,
         supports_forward_ad=True,
     ))
+
+
+def sample_inputs_index_put(op_info, device, dtype, requires_grad, **kwargs):
+    make_arg = partial(make_tensor, dtype=dtype, device=device, requires_grad=requires_grad)
+    make_idx = partial(make_tensor, dtype=torch.long, device=device, requires_grad=False)
+    S = 5
```

Comment:
index_put is a pretty core operation in PyTorch. Could we add some more tests? Maybe a manual test like https://github.com/pytorch/functorch/blob/ec69443bdc1c5ee6b1e143383dc20bb392776597/test/test_vmap.py#L3353-L3364 is sufficient

Reply:
I think the reason for having manual tests for the in place variant is because we want `self` to always have a bdim. With out-of-place `index_put`, we can use `get_fallback_and_vmap_exhaustive` to generate variety of batched inputs. So will try to add more cases in the OpInfo sample.